core/templates/dev/head/domain/utilities/language-util.service.spec.ts

```diff
@@ -114,11 +114,20 @@ describe('Language util service', function() {
 
   it('should correctly check if language supports autogenerated audio',
     () => {
+      let tryCount = 0;
       // Chrome loads voices asynchronously.
       // https://stackoverflow.com/questions/21513706/
+      // Chrome also calls this multiple times and not all languages are
+      // populated the first time.
       window.speechSynthesis.onvoiceschanged = () => {
-        expect(lus.supportsAutogeneratedAudio('hi')).toEqual(false);
-        expect(lus.supportsAutogeneratedAudio('en')).toEqual(true);
+        tryCount++;
+
+        if (window.speechSynthesis.getVoices().length > 0) {
```

Comment:
Perhaps add confirmation that this actually does eventually end up being > 0?

Reply:
I'm hesitant to do that because if an expect fails on the first pass, the test will fail (assuming that it'll take 2 runs to load the languages).

Reply:
Mm. My concern is that this test didn't catch the previous (and current) error in browser-checker.service.ts that was introduced in this PR. If you can find any alternative test that fails when those errors are made, and succeeds with the correct implementation, then that would be fine too.

Reply:
I used `done` so that guarantees that it'll actually run.

Reply:
Used `done`